### ISSUE TITLE
Auth0 - React Native - Android (release app) - weird behaviour of webAuth Login / logout 

### ISSUE THREAD

**Issue by ZubairKh** [auth0/react-native-auth0#392]
Hey everyone!

We have a live app that our users are using! Please need your attention on this problem as quick as possible.

### The Problem 

We have implemented auth0 authentication login into our mobile app built using React Native. We're using webAuth to login (that opens up the webview by auth0 and shows the login form). Also, we did register 2 apps on the auth0 dashboard, one is for our development environment or for testing purposes, another one is for production releaase. 

Our apps are already published on Play and Apple stores and everything seems working fine on both Android and iOS, we were using **`react-native-auth0`** version **`2.6.0`** then we did try to update our app to Play and Apple stores. While making the .aab for Android we were getting the error Manifest merger failed and details error logs were saying we should add **`manifestPlaceholders []`** in build.gradle(app) file. After adding those our build got successful and went live on both the stores. 

Then our QAs noticed on the Android side the login webAuth behaving weirdly like when tapping on the Login button our code was like 

```
const auth0 = new Auth0({
  domain: our_app_name.eu.auth0.com,
  clientId: client_id_from_auth0_dashboard,
})
auth0.webAuth
        .authorize(
          { scope: 'openid email profile' },
          {ephemeralSession: true}
        ).then((credentials) => {
          auth0.auth
            .userInfo({ token: credentials.accessToken })
            .then((data) => {
             // saving data and navigating to Home Page
            })
            .catch((error) => {
              // error logs...
            })
        })
        .catch((error) => {
          // error logs...
        })
```

and we have to tap 3 times in order for this login code to work. On the first tap, we do enter login credentials on webview opened by auth0, then on second tap we see empty webview and disappear after 1 second, third tap same as second and we get a successful response in .then() section of webAuth. The same goes with Logout scenarios like 

 ```
auth0.webAuth
              .clearSession()
              .then((success) => {
                // remove local storage and navigating to Login
              })
              .catch((error) => {
                // error
              })
```

We read the auth0 latest documentation that said 

> Before version 2.9.0, this SDK required you to add an intent filter to the Activity on which you're going to receive the authentication result, and to use the **`singleTask`** **launchMode** in that activity. To migrate your app to version 2.9.0+, remove both and continue with the instructions below.

So we updated the version to 2.9.0, removed intent-filters, and added manifest placeholders as suggested in docs

```
android {
    defaultConfig {
        // Add the next line
        manifestPlaceholders = [auth0Domain: "YOUR_AUTH0_DOMAIN", auth0Scheme: "${applicationId}"]
    }
    ...
}
```
The development environment working great on both iOS and Android. But when we published the app on Android got an issue while tapping on the Login button, the auth0 webview showing empty. I was able to get the inspect element of that webview call and console error saying 

`Navigation not reachable our_app_package://our_app.eu.auth0.com/android/our_app_package/callback`

Everything working fine on 
- Development Environment (iOS, Android) both
- Production Environment (iOS working perfectly) Only Android got these issues

We already checked the auth0 dashboard, everything was set up correctly like **Allowed Callback URLs**, **Allowed Logout URLs** for both iOS and Android (Dev/Prod environments). Previously working fine these issues happened suddenly. Also for saving time in your investigation, we have checked capital or small letters thing in app name, package name and everything is correct as expected on the auth0 dashboard for both dev and prod environments.

The last thing we want to mention is when we put back **`intent-filters`** (along with **`manifestPlaceholders`**) in **`version 2.9.0`**, the app got the similar behaviour that we told you in the start like **(3 times tap on the button for login/logout)**

### Expected behavior?

The webAuth login should work first time in auth0 (version 2.9.0) webview 

### Reproduction

- Step 1: Install the latest version 2.9.0
- Step 2: using auth0.webAuth() on login button press
- Step 2: Run release (Play store app) and press the login button

### Environment

- **Version 2.6.0 & 2.9.0**
- **Framework: React native**

Hope for the quick response, as our user base is growing day by day and this particular error letting our Android users down on our app.

Thanks

**Comment by Widcket** [comment]
Hi @ZubairKh I'm looking into this, will update later today.

**Comment by ZubairKh** [comment]
@Widcket Thanks for putting my attention towards Merged Manifest. Yes, both the activities have mentioned with correct values but I've got the hint from here because we're using multiple product flavors in our app like in the **dev** environment we have **.dev** at the end of the package name while for the **prod** environment we have a **different package name**. 

I'll play around with it by changing the Build Variants and check the Merged Manifest to see what values we got there.

Thanks again for the useful hint, I'll get back hopefully with some resolution.

**Comment by ZubairKh** [comment]
@Widcket Just resolved my issue. I'm explaining here so anybody who will go through the same issue can fix this

Actually, in the case of multiple product flavors we should be adding **`manifestPlaceholders:  []`** multiple times for each flavor. I was only adding manifestPlaceholders inside the android object in `build.gradle` file like mentioned in your documentation

```
android {
    defaultConfig {
        manifestPlaceholders = [auth0Domain: "YOUR_AUTH0_DOMAIN", auth0Scheme: "${applicationId}"]
    }
    ...
}
```

But for different flavors, if we have different package names then we have to give manifest placeholders for each of the flavors like this

```
flavorDimensions "default"

    def auth0Domain = "vidivet.eu.auth0.com"

    productFlavors {
        dev {
            applicationId "com.appName.dev"
            manifestPlaceholders = [auth0Domain: auth0Domain, auth0Scheme: "${applicationId}"]
        }
        prod {
            applicationId "com.appName"
            manifestPlaceholders = [auth0Domain: auth0Domain, auth0Scheme: "${applicationId}"]
        }
    }
```

As we got two different packages for dev and prod, for some reason adding manifestPlaceholders only under android object wasn't picking up the prod flavor `applicationId` but instead was picking up the default `applicationId` under `defaultConfig` object. 

@Widcket my suggestion would be to give this piece of information in your official documentation like in case of multiple flavors if the package name is different for it then we should add manifestPlaceholders for each flavor

Thanks again @Widcket your hint regarding Merged Manifest helped me reach this solution.

Closing this now!
